fluent-gecko/src/dom_localization.js

```diff
@@ -1,4 +1,24 @@
+/* global L10nRegistry */
 import DOMLocalization from '../../fluent-dom/src/dom_localization';
 
-this.DOMLocalization = DOMLocalization;
+/**
+ * The default localization strategy for Gecko. It comabines locales
+ * available in L10nRegistry, with locales requested by the user to
+ * generate the iterator over MessageContexts.
+ *
+ * In the future, we may want to allow certain modules to override this
+ * with a different negotitation strategy to allow for the module to
+ * be localized into a different language - for example DevTools.
+ */
+function defaultGenerateMessages(resourceIds) {
+  return L10nRegistry.generateContexts(undefined, resourceIds);
+}
+
+class GeckoDOMLocalization extends DOMLocalization {
+  constructor(resourceIds, generateMessages = defaultGenerateMessages) {
```

Comment:
Is there a reason to allow passing custom `generateMessages` to the constructor of `GeckoDOMLocalization`? It would appear that it will always want to use the L10nRegistry.

If it's for testing, then perhaps we should stub `L10nRegistry.generateContexts` rather than increase the API surface?

Reply:
Sure, done.

Reply:
ugh, actually, after trying it on tests, it's pretty quirky to do. I think I'd prefer to expose the ability to override `generateMessages` on GeckoLocalization/GeckoDOMLocalization.